linux-kernel: update to 6.12.9

KexyBiscuit opened the pull request:
Topic Description
-----------------

- kernel-base: enable building and shipping linux+kernel on Loongson 3
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: update to 6.12.9
    - Synchronize patches from linux-kernel.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux+kernel: update to 6.12.9
    - Enable building and shipping on Loongson 3.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: update to 6.12.9
    - Routine patch due diligence for upstream releases.
    - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
    - Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am-shazam.html>.
    - Refresh patch series that have been updated upstream.
    - Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.12, current HEAD is 5b2272d31d16d05aabd9de83db580addd9d6f712.
    - Track Phytium patches at https://github.com/AOSC-Tracking/phytium-linux-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
    - Track Ubuntu patches at https://github.com/AOSC-Tracking/ubuntu-kernel-next-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
    - Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is f65a1ce63e2b15af8fb76cdb725bef676e2bf1f9.
    - Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is 02769096e4bb494ebda9dab662ca44ba4bb7ef63.
    - Track linux-surface patches at https://github.com/AOSC-Tracking/linux-surface-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
    - Track loongarch-xe patches at https://github.com/AOSC-Tracking/FanFansfan-loongson-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.12.9, current HEAD is 957600bd0e6f7ba19e66d333c787d88f651f8fa3.
    - Enable building and shipping on Loongson 3.
    - Backport patches for amd-pstate driver to fix performance issues.
    - Add AMD 3D V-Cache optimizer driver.
    - Add patches for Loongson I²S audio devices.
    - Add patches to support DRM Panic on nouveau.
    - Fix comment about amdgpu.abmlevel defaults.
    - Backport LoongArch KVM patches from 6.13.
    - Add patches to fix touchpad on Lenovo ThinkPad L15 Gen 4 and similar devices.
    - Update NT synchronization primitive driver.
    - Add patch to fix issues on Loongson 3A4000 caused by HAVE_UNSTABLE_SCHED_CLOCK.
    - Add patches to fix LEFI boardinfo incorrectly showing ROM Size by removing it.
    - Add patches to support DRM Panic on amdgpu.
    - Add patch to disable CRASH_DUMP on MIPS by default.
    - Add patches to support LEDs on Lenovo laptops.
    - Add patch to fix warnings during S3 suspend on LoongArch.
    - Add patch to fix building of mips_ejtag_fdc driver.
    - Add MIPS patches from Lemote, rebased to 6.12+ by Mingcong Bai.
    - Add patch from Debian, use RELAXED ieee754 mode for Loongson-3 as 3A 4000 is 2008-only.
    - Update linux-surface patches.
    - Update patch from kernel_compiler_patch.
    - Temporarily disable Loongson3 LL/SC errata check to fix building.
    - Add ls7a1000_support module parameter to help users of LS7A2000/2K2000-based hardware leverage the loongson DRM driver.
    - Mark 8BAB board for OMEN thermal profile in hp-wmi driver.
    - MIPS: Check address space in ADE.
    - Fix if branch syntax in ideapad-laptop driver.
    - Disable enable_virt_at_load in KVM by default to fix desktop virtualization solutions like VirtualBox.
    - Reset LSM to AppArmor's default.
    - Enable DRM_LOONGSON on LoongArch.
    - For more details, see commits at e623832ebc9a3173a3e8fd386a24375a56146863, like <https://github.com/AOSC-Dev/aosc-os-abbs/commits/e623832ebc9a3173a3e8fd386a24375a56146863>.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

Package(s) Affected
-------------------

- kernel-base: 5
- kernel-tools: 6.12.9-3
- linux+kernel: 3:6.12.9
- linux-kernel-${__VER}: 1:6.12.9-3

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools kernel-base
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2 Hyper-V VM.

![image](https://github.com/user-attachments/assets/bcbcdc2e-1086-475f-a9d5-600d784540ac)

KexyBiscuit commented:
FTBFS on Armv8-A AArch64 with `CONFIG_NFS_LOCALIO=y`.

```
ld: fs/nfs/localio.o: in function `nfs_local_iocb_alloc':
/var/cache/acbs/build/acbs.um5811_i/linux-6.12-rc1/fs/nfs/localio.c:290:(.text+0x3d4): undefined reference to `nfs_to'
ld: fs/nfs/localio.o: relocation R_AARCH64_ADR_PREL_PG_HI21 against symbol `nfs_to' which may bind externally can not be used when making a shared object; recompile with -fPIC
/var/cache/acbs/build/acbs.um5811_i/linux-6.12-rc1/fs/nfs/localio.c:290:(.text+0x3d4): dangerous relocation: unsupported relocation
ld: /var/cache/acbs/build/acbs.um5811_i/linux-6.12-rc1/fs/nfs/localio.c:290:(.text+0x3e0): undefined reference to `nfs_to'
```

<https://lore.kernel.org/all/202410021451.Dj3s8Owx-lkp@intel.com/>

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.1 build-24319023
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27749.1000) 10.0.27749
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/c583d9f6-7b70-4abc-9d9b-4a380d848b1b)

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2, AMD64 generation 1 and Armv8-A generation 2 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/0e394dbc-5054-4f69-8c97-dd8a023836a0)
![image](https://github.com/user-attachments/assets/4704335f-2cac-405f-bf5a-36968cb8395c)
![image](https://github.com/user-attachments/assets/248ea91a-c617-4806-92f8-02e87468d33d)

KexyBiscuit commented:
Smoke testing passed on XA612A0.

![图片](https://github.com/user-attachments/assets/11fdb989-30f1-4c1c-9088-17465725c26d)

KexyBiscuit commented:
Smoke testing passed on Ampere A110 Raptor.

![image](https://github.com/user-attachments/assets/d7cc7e3b-b685-4056-98a4-a400930e16ea)

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.1 build-24319023
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27749.1000) 10.0.27749
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/841d0b46-b771-41a2-8f92-ba73ec75deda)

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2, AMD64 generation 1 and Armv8-A generation 2 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/11681694-1985-4f92-b93b-3341bcaba282)
![image](https://github.com/user-attachments/assets/ea38ad50-d482-4192-b9d9-61df727b4005)
![image](https://github.com/user-attachments/assets/77603e38-c3e1-4199-a466-8e47d28a336f)

KexyBiscuit commented:
Smoke testing passed on Ampere A110 Raptor.

![image](https://github.com/user-attachments/assets/ba891aa6-1b81-498d-bfa2-5de7fa9ead97)

KexyBiscuit commented:
Smoke testing passed on LX-2510.

![image](https://github.com/user-attachments/assets/4f6c36f1-590f-4f91-b051-8bfa4f0445ec)

jiegec commented:
Tested on vmware:

```
               gpBBBBBBBBBP                   -----------------
           _gBBBBBBBBBRP                      OS: AOSC OS (amd64)
         4BBBBBBBBRP  ,_____                  Host: VMware7,1 None
              `"" _g@@@@@@@@@@@@@%g>          Kernel: 6.12.0-aosc-main
              __@@@@@@@@@@@@@@@@P"  ___       Uptime: 45 secs
           _g@@@@@@@@@@@@@@@N"` _gN@@@@@N^    Packages: 5 (cargo), 2 (npm), 2680 (dpkg)
       _w@@@@@@@@@@@@@@@@P" _g@@@@@@@P"       Shell: fish 3.7.1
    _g@@@@@@@@@@@@@@@N"`  VMNN@NNNM^`         Resolution: 1280x768
  ^MMM@@@@@@@@@@@MP" ,ggppww__                Terminal: /dev/pts/0
          `""""" _wNNNNNNNNNNNNNNNNNNN        CPU: AMD EPYC 7H12 (64) @ 2.5GHz
              _gBNNNNNNNNNNNNNNNNNP"          GPU: VMware SVGA II Adapter
          _wNNNNNNNNNNNNNNNNNNMP`             Memory: 4.64 GiB / 503.58 GiB (0%)
       _gBNNNNNNNNNNNNNNNNNP"                 Network: 10 Gbps
   _wNNNNNNNNNNNNNNNNNNNM^                    BIOS: VMware, Inc. (05/22/2023)
   ""Y^^MNNNNNNNNNNNNP`
           `"""""""



```

jiegec commented:
THP performance issue fixed, see https://github.com/jiegec/blog-source/commit/d045df9a1b5efb8c3eba46713b0e32c9f8bed537

KexyBiscuit commented:
Smoke testing passed on XA612A0.

![图片](https://github.com/user-attachments/assets/8d3fa387-084d-459e-a4df-0d7daba54a40)

KexyBiscuit commented:
Smoke testing passed on Latitude D630.

![image_2024-11-20_14-55-07](https://github.com/user-attachments/assets/c8571960-2704-463a-b9de-571cdab47fe3)

KexyBiscuit commented:
Smoke testing passed on Ampere A110 Raptor.

![image](https://github.com/user-attachments/assets/ea0d084a-388c-4fff-876f-ec377206a218)

KexyBiscuit commented:
Smoke testing passed on AMD64 generation 2, AMD64 generation 1 and Armv8-A generation 2 Hyper-V VMs.

![image](https://github.com/user-attachments/assets/c038cf2b-26a6-450d-8dfd-668870c776ba)
![image](https://github.com/user-attachments/assets/247cac96-642b-4187-a04f-cfe54e830889)
![image](https://github.com/user-attachments/assets/e49f2a97-e1b9-4543-ac13-ff906ff51094)

KexyBiscuit commented:
Smoke testing passed on LX-2510.

![image](https://github.com/user-attachments/assets/fa43392d-a05d-4e9a-bdab-94b16215f154)

KexyBiscuit commented:
Smoke testing passed on AMD64 VMware VM.

- Product: VMware® Workstation 17 Pro
- Version: 17.6.2 build-24409262
- Host OS version: Windows 11 Enterprise, 64-bit (Build 27766.1000) 10.0.27766
- Accelerate 3D graphics: on
- Firmware type: UEFI
- Host VBS Mode: on

![image](https://github.com/user-attachments/assets/cacdcb5d-0847-4c7a-96ae-4c94977f0534)

KexyBiscuit committed:
linux-kernel: update to 6.12.9

- Routine patch due diligence for upstream releases.
  - Now following contribution guidelines of Android common kernels <https://android.googlesource.com/kernel/common/+/refs/heads/android-mainline/README.md>.
  - Now use B4 shazam to apply patches from list <https://b4.docs.kernel.org/en/latest/maintainer/am-shazam.html>.
  - Refresh patch series that have been updated upstream.
  - Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.12, current HEAD is 5b2272d31d16d05aabd9de83db580addd9d6f712.
  - Track Phytium patches at https://github.com/AOSC-Tracking/phytium-linux-kernel @ aosc/v6.6, current HEAD is 3ea48e9db5214367ad77e786ea91ca6d68227af6.
  - Track Ubuntu patches at https://github.com/AOSC-Tracking/ubuntu-kernel-next-linux @ aosc/v6.11, current HEAD is 3c088ba346d5dc0a2861039fee7834626f9eec88.
  - Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is f65a1ce63e2b15af8fb76cdb725bef676e2bf1f9.
  - Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is 02769096e4bb494ebda9dab662ca44ba4bb7ef63.
  - Track linux-surface patches at https://github.com/AOSC-Tracking/linux-surface-kernel @ aosc/v6.12, current HEAD is aee628d0f5e6346e83fe00f5614bb60a7c62cee9.
  - Track loongarch-xe patches at https://github.com/AOSC-Tracking/FanFansfan-loongson-linux @ aosc/v6.8, current HEAD is 565064d2079323994aa041f449649918513c7b03.
  - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.12.9, current HEAD is 056e094f8eac3f904122d2ae4fef666e5fff6906.
- Enable building and shipping on Loongson 3.
- Backport patches for amd-pstate driver to fix performance issues.
- Add AMD 3D V-Cache optimizer driver.
- Add patches for Loongson I²S audio devices.
- Add patches to support DRM Panic on nouveau.
- Fix comment about amdgpu.abmlevel defaults.
- Backport LoongArch KVM patches from 6.13.
- Add patches to fix touchpad on Lenovo ThinkPad L15 Gen 4 and similar devices.
- Update NT synchronization primitive driver.
- Add patch to fix issues on Loongson 3A4000 caused by HAVE_UNSTABLE_SCHED_CLOCK.
- Add patches to fix LEFI boardinfo incorrectly showing ROM Size by removing it.
- Add patches to support DRM Panic on amdgpu.
- Add patch to disable CRASH_DUMP on MIPS by default.
- Add patches to support LEDs on Lenovo laptops.
- Add patch to fix warnings during S3 suspend on LoongArch.
- Add patch to fix building of mips_ejtag_fdc driver.
- Add MIPS patches from Lemote, rebased to 6.12+ by Mingcong Bai.
- Add patch from Debian, use RELAXED ieee754 mode for Loongson-3 as 3A 4000 is 2008-only.
- Update linux-surface patches.
- Update patch from kernel_compiler_patch.
- Temporarily disable Loongson3 LL/SC errata check to fix building.
- Add ls7a1000_support module parameter to help users of LS7A2000/2K2000-based hardware leverage the loongson DRM driver.
- Mark 8BAB board for OMEN thermal profile in hp-wmi driver.
- MIPS: Check address space in ADE.
- Fix if branch syntax in ideapad-laptop driver.
- Disable enable_virt_at_load in KVM by default to fix desktop virtualization solutions like VirtualBox.
- Reset LSM to AppArmor's default.
- Enable DRM_LOONGSON on LoongArch.
- For more details, see commits at e623832ebc9a3173a3e8fd386a24375a56146863, like <https://github.com/AOSC-Dev/aosc-os-abbs/commits/e623832ebc9a3173a3e8fd386a24375a56146863>.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux+kernel: update to 6.12.9

- Enable building and shipping on Loongson 3.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: update to 6.12.9

- Synchronize patches from linux-kernel.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-base: enable building and shipping linux+kernel on Loongson 3

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai commented:
Ship it!
